### ISSUE TITLE
Cache the CID

### ISSUE THREAD

**Issue by Stebalien** [ipfs/go-ipld-eth#3]
It's probably a good idea to cache the CID (and, if possible the output of RawData) as these methods are often called repeatedly in fast-path code.

**Comment by hermanjunge** [comment]
Part of the (_my_ TBH) plan is to take advantage at maximum of the cache in this module, as you know, the state trie elements requests will have a lot of common elements.

Please, @Stebalien, point me to some place in the IPFS code stack where cache is being used, in order to leverage that knowledge.

**Comment by Stebalien** [comment]
> Part of the (my TBH) plan is to take advantage at maximum of the cache in this module, as you know, the state trie elements requests will have a lot of common elements.

Could you rephrase this? I'm not sure I understand your question. It sounds like you're talking about some form of inter-node caching which would usually be done by the DAGService/Blockservice (although the DAGService doesn't currently cache anything).

---

By cache, I mean cache *within* the node. Really, the best way to do this is to just precompute the CID when constructing the node.

**Comment by hermanjunge** [comment]
I see. I was thinking on `DAGService/Blockservice` cache.

**Comment by hermanjunge** [comment]
Done [here](https://github.com/ipfs/go-ipld-eth/pull/5). Each `cid` is computed once in the object lifetime.
